shared/ui/Stream/PullRequests/Bitbucket/PullRequest.tsx

```diff
@@ -491,377 +249,533 @@ export const PullRequest = () => {
 		}
 	};
 
 	/**
-	 * Called after an action that requires us to re-fetch from the provider
+	 * This is called when a user clicks the "reload" button.
+	 * with a "hard-reload" we need to refresh the conversation and file data
 	 * @param message
 	 */
-	const fetch = async (message?: string) => {
+	const reload = async (message?: string) => {
+		console.log("PullRequest is reloading");
 		if (message) setIsLoadingMessage(message);
 		setIsLoadingPR(true);
-
 		const response = await dispatch(
 			getPullRequestConversationsFromProvider({
 				providerId: derivedState.currentPullRequestProviderId!,
 				id: derivedState.currentPullRequestId!,
 			})
 		).unwrap();
-		_assignState(response);
-	};
-
-	const reload = async (message?: string) => {
-		console.log("MergeRequest is reloading");
-		fetch(message);
+		_assignState(response, "reload");
 
 		// just clear the files and commits data -- it will be fetched if necessary (since it has its own api call)
 		dispatch(
-			clearPullRequestFiles({
-				providerId: derivedState.currentPullRequestProviderId!,
-				id: derivedState.currentPullRequestId!,
-			})
+			clearPullRequestFiles(
+				derivedState.currentPullRequestProviderId!,
+				derivedState.currentPullRequestId!
+			)
 		);
 		dispatch(
 			clearPullRequestCommits({
 				providerId: derivedState.currentPullRequestProviderId!,
 				id: derivedState.currentPullRequestId!,
 			})
 		);
-		// we can force the child components to update
-		// by changing part of its key attribute
-		setDynamicKey(new Date().getTime().toString());
 	};
 
-	const __onDidRender = functions => {
-		insertText = functions.insertTextAtCursor;
-		insertNewline = functions.insertNewlineAtCursor;
-		focusOnMessageInput = functions.focus;
+	const checkout = async () => {
+		if (!pr) return;
+
+		setIsLoadingBranch(true);
+
+		const repoId = derivedState.prRepoId || "";
+		const result = await HostApi.instance.send(SwitchBranchRequestType, {
+			branch: pr!.headRefName,
+			repoId: repoId,
+		});
+		if (result.error) {
+			logError(result.error, {
+				prRepoId: derivedState.prRepoId,
+				branch: pr.headRefName,
+				repoId: repoId,
+				prRepository: pr!.repository,
+			});
+
+			confirmPopup({
+				title: "Git Error",
+				className: "wide",
+				message: (
+					<div className="monospace" style={{ fontSize: "11px" }}>
+						{result.error}
+					</div>
+				),
+				centered: false,
+				buttons: [{ label: "OK", className: "control-button" }],
+			});
+			setIsLoadingBranch(false);
+		} else {
+			setIsLoadingBranch(false);
+			getOpenRepos();
+		}
+	};
+
+	useEffect(() => {
+		if (pr && pr.headRefName && derivedState.checkoutBranch) {
+			checkout();
+			// clear the branch flag
+			dispatch(setCurrentPullRequest(pr.providerId, pr.id));
+		}
+	}, [pr && pr.headRefName, derivedState.checkoutBranch]);
+
+	useEffect(() => {
+		if (!pr) return;
+
+		const _didChangeDataNotification = HostApi.instance.on(
+			DidChangeDataNotificationType,
+			(e: any) => {
```

Comment:
This is auto-typed - can remove the `any` and it becomes a DidChangeNotification

```suggestion
			(e) => {
```